web/src/lib/data-hooks/useRoadmap.ts

```diff
@@ -31,8 +31,10 @@ export const useRoadmap = (): UseRoadmapReturnValue => {
     return [...new Set(sections)];
   }, [roadmap]);
 
+  const isRoadMapEmpty = roadmap?.steps.length === 0 || roadmap?.tasks.length === 0;
+
   useMountEffectWhenDefined(() => {
-    if (!roadmap) {
+    if (!roadmap || isRoadMapEmpty) {
```

Comment:
Given this is the only place where you use the variable `isRoadMapEmpty`, I'd love to see you combine all your conditionals into one:

```typescript
const roadmapNeedsRebuild = !roadmap || roadmap.steps.length === 0 || roadmap.tasks.length === 0;
```

(Plus, this allows you to drop the optionals in your conditional.)

That way, your branching statement reads a little better:

```typescript
if (roadmapNeedsRebuild) {
  buildAndSetRoadmap();
}
```

(I'm totally open to better names, btw.)

Reply:
@seidior Not sure if we've talked about this at all, but we've been trying to label our code comments with a weight. We've been using a [modified version of the rock scale](https://docs.google.com/document/d/1aTAQthQf704Ed0hOEDAdU7Vq41B_YrwraiiU05PugYY/edit#heading=h.njm8kqnslu7y) to do this. The idea is to help the reviewer consider and the reviewee to understand how important a given comment is.

Ex: [pebble] Suggestion to follow.

Reply:
Going with `rebuildRoadmap` as the variable name.